drivers/unix/net_socket_posix.cpp

```diff
@@ -306,7 +306,7 @@ Error NetSocketPosix::bind(IP_Address p_addr, uint16_t p_port) {
 	sockaddr_storage addr;
 	size_t addr_size = _set_addr_storage(&addr, p_addr, p_port, _ip_type);
 
-	if (::bind(_sock, (struct sockaddr *)&addr, addr_size) == SOCK_EMPTY) {
+	if (::bind(_sock, (struct sockaddr *)&addr, addr_size) == (int)SOCK_EMPTY) {
```

Comment:
Okay, this should actually be:
`if (::bind(_sock, (struct sockaddr *)&addr, addr_size) != 0)`
It does not return a socket, but 0 on success, -1 on failure (actually `SOCKET_ERROR` on windows -.-).
Not sure why I used `SOCK_EMTPY` there originally, but it's wrong.

Reply:
Thanks, fixed.